### PR TITLE
Od svg

### PR DESCRIPTION
**Pull request by rohanakut**
Their PR is wrt to #441 . The handler creates a bounding box for every category of the object. The following testing practices were used for testing this code

1. Image used for testing can be found [here](https://github.com/Shared-Reality-Lab/IMAGE-test-graphics/blob/main/photos/0000/cathedral.jpg)
2. The following figure shows the output of the handler when the object detection detects the benches in the image3. 
Note: In the following images only the benches have been highlighted. There are multiple different objects detected by the YOLO model, but for ease of understanding only the benches have been shown

Screen Shot 2022-08-21 at 3 43 01 PM

3. The following was the json response of the handler which was tested on unicorn
```
{'request_uuid': '88a10b92-e1a9-4a17-b8a9-695934243bcb', 
'timestamp': 1661110845, 
'renderings': [{'type_id': 'ca.mcgill.a11y.image.renderer.SVGLayers',
 'description': 'SVG visualization', 
'data': {
'layers': [{'label': 'person ', 'svg': 'data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz4KPHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHhtbG5zOnhsaW5rPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5L3hsaW5rIgogICAgIHdpZHRoPSI2ODQiIGhlaWdodD0iNDU2IiB2aWV3Qm94PSIwIC00NTYgNjg0IDQ1NiI+CjxkZWZzPgo8L2RlZnM+CjxyZWN0IHg9IjYwMiIgeT0iLTI4OCIgd2lkdGg9IjQ1IiBoZWlnaHQ9IjUwIiBzdHJva2U9IiNmZjQ0NzciIGZpbGwtb3BhY2l0eT0iMCIgLz4KPHJlY3QgeD0iMTU3IiB5PSItMzQxIiB3aWR0aD0iNDEiIGhlaWdodD0iNTQiIHN0cm9rZT0iI2ZmNDQ3NyIgZmlsbC1vcGFjaXR5PSIwIiAvPgo8cmVjdCB4PSI1MDgiIHk9Ii0zNTMiIHdpZHRoPSI0NSIgaGVpZ2h0PSI3OSIgc3Ryb2tlPSIjZmY0NDc3IiBmaWxsLW9wYWNpdHk9IjAiIC8+CjxyZWN0IHg9IjQyNiIgeT0iLTMzNCIgd2lkdGg9IjQ1IiBoZWlnaHQ9Ijg0IiBzdHJva2U9IiNmZjQ0NzciIGZpbGwtb3BhY2l0eT0iMCIgLz4KPHJlY3QgeD0iMzcwIiB5PSItMzM3IiB3aWR0aD0iNjciIGhlaWdodD0iODYiIHN0cm9rZT0iI2ZmNDQ3NyIgZmlsbC1vcGFjaXR5PSIwIiAvPgo8cmVjdCB4PSI1MzMiIHk9Ii0zMzAiIHdpZHRoPSI2OSIgaGVpZ2h0PSI4OSIgc3Ryb2tlPSIjZmY0NDc3IiBmaWxsLW9wYWNpdHk9IjAiIC8+CjxyZWN0IHg9IjM0OSIgeT0iLTI4NyIgd2lkdGg9IjM1IiBoZWlnaHQ9IjEwMSIgc3Ryb2tlPSIjZmY0NDc3IiBmaWxsLW9wYWNpdHk9IjAiIC8+CjxyZWN0IHg9IjI3NiIgeT0iLTMyNyIgd2lkdGg9IjQwIiBoZWlnaHQ9IjEzNyIgc3Ryb2tlPSIjZmY0NDc3IiBmaWxsLW9wYWNpdHk9IjAiIC8+CjxyZWN0IHg9IjAiIHk9Ii0xNTMiIHdpZHRoPSI2NSIgaGVpZ2h0PSIxNTEiIHN0cm9rZT0iI2ZmNDQ3NyIgZmlsbC1vcGFjaXR5PSIwIiAvPgo8cmVjdCB4PSIxNDQiIHk9Ii0xMzUiIHdpZHRoPSIxNTMiIGhlaWdodD0iMTMzIiBzdHJva2U9IiNmZjQ0NzciIGZpbGwtb3BhY2l0eT0iMCIgLz4KPHJlY3QgeD0iMCIgeT0iLTM5OSIgd2lkdGg9IjQ3IiBoZWlnaHQ9IjE3MCIgc3Ryb2tlPSIjZmY0NDc3IiBmaWxsLW9wYWNpdHk9IjAiIC8+CjxyZWN0IHg9IjYwMyIgeT0iLTQ1MCIgd2lkdGg9Ijc5IiBoZWlnaHQ9IjE2NiIgc3Ryb2tlPSIjZmY0NDc3IiBmaWxsLW9wYWNpdHk9IjAiIC8+CjxyZWN0IHg9IjQ0NiIgeT0iLTMyMCIgd2lkdGg9Ijg5IiBoZWlnaHQ9IjE3MiIgc3Ryb2tlPSIjZmY0NDc3IiBmaWxsLW9wYWNpdHk9IjAiIC8+CjxyZWN0IHg9Ijk1IiB5PSItMzI2IiB3aWR0aD0iNTkiIGhlaWdodD0iMTgwIiBzdHJva2U9IiNmZjQ0NzciIGZpbGwtb3BhY2l0eT0iMCIgLz4KPHJlY3QgeD0iNjQyIiB5PSItMzE0IiB3aWR0aD0iNDIiIGhlaWdodD0iMTg5IiBzdHJva2U9IiNmZjQ0NzciIGZpbGwtb3BhY2l0eT0iMCIgLz4KPHJlY3QgeD0iMCIgeT0iLTI4NSIgd2lkdGg9IjYwIiBoZWlnaHQ9IjE5NyIgc3Ryb2tlPSIjZmY0NDc3IiBmaWxsLW9wYWNpdHk9IjAiIC8+CjxyZWN0IHg9IjM4OSIgeT0iLTI1NCIgd2lkdGg9IjI5NCIgaGVpZ2h0PSIyNTAiIHN0cm9rZT0iI2ZmNDQ3NyIgZmlsbC1vcGFjaXR5PSIwIiAvPgo8cmVjdCB4PSIxMjciIHk9Ii0zNzYiIHdpZHRoPSIxNzAiIGhlaWdodD0iMzMxIiBzdHJva2U9IiNmZjQ0NzciIGZpbGwtb3BhY2l0eT0iMCIgLz4KPHJlY3QgeD0iMzgiIHk9Ii0zNjMiIHdpZHRoPSIxMTciIGhlaWdodD0iMzU3IiBzdHJva2U9IiNmZjQ0NzciIGZpbGwtb3BhY2l0eT0iMCIgLz4KPHJlY3QgeD0iMzAxIiB5PSItMzk1IiB3aWR0aD0iMTkyIiBoZWlnaHQ9IjM4OCIgc3Ryb2tlPSIjZmY0NDc3IiBmaWxsLW9wYWNpdHk9IjAiIC8+Cjwvc3ZnPg=='}]}}]}
```

### CODE REVIEW

File: handlers/svg-object-detection/od_svg.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        )
+        validator.validate(contents)
+    except ValidationError as e:
+        logging.error(e)
+        return jsonify("Invalid request received!"), 400
+
+    # Check preprocessor data
+    preprocessors = contents['preprocessors']
+    print(contents['capabilities'][0])
+    if(contents['capabilities'][0] != "ca.mcgill.a11y.image.capability.DebugMode"):
```
Comment (jaydeepsingh25) [inline handlers/svg-object-detection/od_svg.py:66]: A more robust check would be:
capabilities = contents["capabilities"]
if ("ca.mcgill.a11y.image.capability.DebugMode" not in  capabilities):

File: handlers/svg-object-detection/od_svg.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        logging.debug("Sending response")
+        return response
+    if "dimensions" in contents:
+        # If an existing graphic exists, often it is
+        # best to use that for convenience.
+        # see the following for SVG coordinate info:
+        # developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Positions
+        dimensions = contents["dimensions"]
+
+    objects = preprocessors["ca.mcgill.a11y.image.preprocessor.objectDetection"]["objects"]
```
Comment (jaydeepsingh25) [inline handlers/svg-object-detection/od_svg.py:106]: I guess all the following code should be in if block, what if dimensions is undefined. The code will break at line 109.

Comment (rohanakut) [reply handlers/svg-object-detection/od_svg.py:106]: @jaydeepsingh25 there is already a check happening on line 99, where I am checking if dimensions are present in the input request. I don't think we need an additional check for the same

**Comment by rohanakut** [comment]
@jaydeepsingh25 there has been some significant changes in the logic. There was a dimension issue, which was realized on further testing. Sure to these changes, I need to re-test the code on unicorn. Hence converting this to draft PR

**Comment by jeffbl** [comment]
@jaydeepsingh25 if you get a chance to review this for merge, it would let @rohanakut work on these if any problems arise, before you start working on the renderer for them in the extension.

**Comment by jeffbl** [comment]
@rohanakut please assign back to @jaydeepsingh25 when this is ready for review.

**Comment by rohanakut** [comment]
@jaydeepsingh25 this has been tested on unicorn and is working

**Comment by jeffbl** [comment]
@rohanakut Jaydeep is finishing up the v3 extension work, so will be turning back to vetting PRs. I see you made a couple of commits since Jaydeep assigned this back to you. Is it ready for review? If so, please assign to @jaydeepsingh25

**Pull request merged**